plugins/builds.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+
+const joi = require('joi');
+const boom = require('boom');
+
+const SCHEMA_BUILD_ID = joi.string().hex().length(40).label('Build ID');
+const SCHEMA_ARTIFACT_ID = joi.string().label('Artifact ID');
+const DEFAULT_TTL = 24 * 60 * 60 * 1000; // 1 day
+const DEFAULT_BYTES = 1024 * 1024 * 1024; // 1GB
+
+/**
+ * Builds Plugin
+ * @method register
+ * @param  {Hapi}     server                Hapi Server
+ * @param  {Object}   options               Configuration
+ * @param  {Integer}  options.expiresIn     How long to keep it around
+ * @param  {Integer}  options.maxByteSize   Maximum Bytes to accept
+ * @param  {Function} next                  Function to call when done
+ */
+exports.register = (server, options, next) => {
+    const cache = server.cache({
+        segment: 'builds',
+        expiresIn: options.expiresIn || DEFAULT_TTL
+    });
+
+    server.expose('stats', cache.stats);
+    server.route([{
+        method: 'GET',
+        path: '/builds/{id}/{artifact*}',
+        config: {
+            description: 'Read build artifacts',
+            notes: 'Get an artifact from a specific build',
+            tags: ['api', 'builds'],
+            handler: (request, reply) => {
+                const buildId = request.params.id;
+                const artifact = request.params.artifact;
+                const id = `${buildId}-${artifact}`;
+
+                cache.get(id, (err, value) => {
+                    if (err) {
+                        return reply(err);
+                    }
+                    if (err || value === null) {
```

Comment:
Because you already check for `err` in the previous check, you can reduce this to:

```
if (value === null) {
    ...
}
```

Or `value` just has to be non-falsy:

```
if (!value) {
    ...
}
```